bvm/ballerina-core/src/main/java/org/ballerinalang/util/tracer/TraceManagerWrapper.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.util.tracer;
+
+import org.ballerinalang.bre.bvm.WorkerExecutionContext;
+import org.ballerinalang.util.tracer.factory.BTracerFactory;
+import org.ballerinalang.util.tracer.factory.NoOpTracerFactory;
+import org.ballerinalang.util.tracer.factory.TracerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.ballerinalang.util.tracer.TraceConstants.TRACER_MANAGER_CLASS;
+import static org.ballerinalang.util.tracer.TraceConstants.TRACE_PREFIX;
+
+import static java.util.Arrays.asList;
+
+/**
+ * {@link TraceManagerWrapper} loads {@link TraceManager} implementation
+ * and wraps it's functionality.
+ *
+ * @since 0.964.1
+ */
+public class TraceManagerWrapper {
+    private static final TraceManagerWrapper instance = new TraceManagerWrapper();
+    private Map<String, List<Tracer>> tracerRegistry = new HashMap<>();
+    private TraceManager manager;
+    private TracerFactory factory;
+
+    private TraceManagerWrapper() {
+        try {
+            Class<?> tracerManagerClass = Class
+                    .forName(TRACER_MANAGER_CLASS)
+                    .asSubclass(TraceManager.class);
+            manager = (TraceManager) tracerManagerClass.newInstance();
+            factory = (manager.isEnabled())
+                    ? new BTracerFactory()
+                    : new NoOpTracerFactory();
+        } catch (Exception e) {
+            factory = new NoOpTracerFactory();
```

Comment:
Shall we log the exception? If we don't log it, then we may not have any clue if there are any error occurs on this block.

Reply:
The idea here was to get rid of java exceptions, and silently fallback if the tracer manager is not available. Besides, this is highly unlikely happens unless the user intentionally removes the jar with tracer manager impl class.